sql: TestLogic_testserver_test_22_1_22_2  flaked due to empty conn string (master, x86 build)

See flake[ here](https://teamcity.cockroachdb.com/viewLog.html?buildId=8222053&buildTypeId=Cockroach_BazelEssentialCi). 

Stack trace: 
```
No pass/skip/fail event found for test
=== RUN   TestLogic_testserver_test_22_1_22_2
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/a0cd2ed7268f1ad03abb8f1d57f5fcc9/logTestLogic_testserver_test_22_1_22_22661895086
    test_log_scope.go:79: use -show-logs to present logs inline
panic: empty connection string

goroutine 94 [running]:
github.com/cockroachdb/cockroach-go/v2/testserver.(*testServerImpl).pollListeningURLFile(0xc0002fc900, 0x0)
	github.com/cockroachdb/cockroach-go/v2/testserver/external/com_github_cockroachdb_cockroach_go_v2/testserver/testserver.go:722 +0x765
github.com/cockroachdb/cockroach-go/v2/testserver.(*testServerImpl).StartNode.func1()
	github.com/cockroachdb/cockroach-go/v2/testserver/external/com_github_cockroachdb_cockroach_go_v2/testserver/testservernode.go:108 +0x2f
created by github.com/cockroachdb/cockroach-go/v2/testserver.(*testServerImpl).StartNode
	github.com/cockroachdb/cockroach-go/v2/testserver/external/com_github_cockroachdb_cockroach_go_v2/testserver/testservernode.go:107 +0xcbb
```


Jira issue: CRDB-23194

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -1264,6 +1264,7 @@ func (t *logicTest) newTestServerCluster(bootstrapBinaryPath string, upgradeBina
 		testserver.StoreOnDiskOpt(),
 		testserver.CockroachBinaryPathOpt(bootstrapBinaryPath),
 		testserver.UpgradeCockroachBinaryPathOpt(upgradeBinaryPath),
+		testserver.PollListenURLTimeoutOpt(120),
 		testserver.AddListenAddrPortOpt(ports[0]),
 		testserver.AddListenAddrPortOpt(ports[1]),
 		testserver.AddListenAddrPortOpt(ports[2]),

--- a/pkg/sql/logictest/testdata/logic_test/testserver_test_22.1_22.2
+++ b/pkg/sql/logictest/testdata/logic_test/testserver_test_22.1_22.2
@@ -1,5 +1,4 @@
 # LogicTest: cockroach-go-testserver-22.1-22.2
-skip 94871
 
 query I
 SELECT 1

--- a/pkg/sql/logictest/testdata/logic_test/testserver_upgrade_node
+++ b/pkg/sql/logictest/testdata/logic_test/testserver_upgrade_node
@@ -1,7 +1,5 @@
 # LogicTest: cockroach-go-testserver-22.2-master
 
-skip 94956
-
 query I
 SELECT 1
 ----

